components/portal/messages/controllers.js

```diff
@@ -94,33 +103,90 @@ define(['angular'], function(angular) {
             .catch(filterMessagesFailure);
           };
 
-
         /**
-         * Separate the message types in scope for child controllers
-         * @param {Object} result
+         * Filters raw messages by group membership and
+         * valid begin/end dates.
+         * Resolves URLs in messages.
+         *
+         * @param {Object} messages
          */
-        var filterMessagesSuccess = function(result) {
-          $scope.seenMessageIds = result[0];
-          var dataMessages = result[1];
-          var groupedMessages = result[2];
-          var filteredMessages = $filter('filterForCommonElements')(
-            groupedMessages,
-            dataMessages
-          );
-          var dateFilteredMessages =
-            $filter('filterByDate')(filteredMessages);
-            $scope.messages =
-              $filter('separateMessageTypes')(dateFilteredMessages);
-          $scope.hasMessages = true;
+        var configureMessages = function(messages) {
+          $q.all([promiseSeenMessageIds(),
+          promiseMessagesByData(messages),
+          promiseMessagesByGroup(messages)])
+          .then(function(result) {
+            $scope.seenMessageIds = result[0];
+            var dataMessages = result[1];
+            var groupedMessages = result[2];
+            var filteredMessages = $filter('filterForCommonElements')(
+              groupedMessages,
+              dataMessages
+            );
+            var dateFilteredMessages =
+              $filter('filterByDate')(filteredMessages);
+            var userViewedMessages =
+              $filter('filterSeenAndUnseen')(dateFilteredMessages,
+                 $scope.seenMessageIds);
+            $scope.messages = userViewedMessages.unseen;
+            $scope.seenMessages = userViewedMessages.seen;
+            $scope.hasMessages = true;
+            $rootScope.$broadcast('messagesConfigured');
+            return result;
+          })
+            .catch(function(error) {
+              $log.warn(error);
+              $scope.showMessagesFeatures = false;
+            });
         };
+        var reconfigureMessages = $rootScope.$on('configureMessages',
+          function(event, data) {
+            configureMessages(allMessages);
+          });
 
         /**
-         * Handle errors that occur while resolving promises to
-         * get notifications
-         * @param {Object} error
+         * Marks a single message as seen.
```

Comment:
"seen"? Or "dismissed"? Looks like maybe we're conflating "seen" and "dismissed" even down in `$scope.seenMessageIds`?